pkg/cmd/auth/shared/gitcredentials/helper_config_test.go

```diff
@@ -0,0 +1,176 @@
+	require.True(t, repoHelper.IsOurs(), "expected the helper to be ours but was %q", repoHelper.Cmd)
+
+	gistHelper, err := hc.ConfiguredHelper("gist.github.com")
+	require.NoError(t, err)
+	require.True(t, gistHelper.IsConfigured(), "expected our helper to be configured")
+	require.True(t, gistHelper.IsOurs(), "expected the helper to be ours but was %q", gistHelper.Cmd)
+}
+
+func TestConfiguredHelperNoPreexistingHelpersConfigured(t *testing.T) {
+	// Given there are no credential helpers configured
+	withIsolatedGitConfig(t)
+
+	// When we check the configured helper for a hostname
+	hc := gitcredentials.HelperConfig{
+		SelfExecutablePath: "/path/to/gh",
+		GitClient:          &git.Client{},
+	}
+	helper, err := hc.ConfiguredHelper("github.com")
+
+	// Then the helper should not be configured and not ours
+	require.NoError(t, err)
+	require.False(t, helper.IsConfigured(), "expected no helper to be configured")
+	require.False(t, helper.IsOurs(), "expected the helper not to be ours but was %q", helper.Cmd)
+}
+
+func TestConfiguredHelperWithPreexistingGlobalHelperConfigured(t *testing.T) {
+	// Given there is a global credential helper configured
+	withIsolatedGitConfig(t)
+	configureTestCredentialHelper(t, "credential.helper")
+
+	// When we check the configured helper for a hostname
+	hc := gitcredentials.HelperConfig{
+		SelfExecutablePath: "/path/to/gh",
+		GitClient:          &git.Client{},
+	}
+	helper, err := hc.ConfiguredHelper("github.com")
+
+	// Then the helper should be configured, but not ours
+	require.NoError(t, err)
+	require.True(t, helper.IsConfigured(), "expected no helper to be configured")
+	require.False(t, helper.IsOurs(), "expected the helper not to be ours but was %q", helper.Cmd)
+}
+
+func TestConfiguredHelperWithPreexistingHostHelperConfigured(t *testing.T) {
+	// Given there is a credential helper configured for a hostname
+	withIsolatedGitConfig(t)
+	configureTestCredentialHelper(t, "credential.https://github.com.helper")
+
+	// When we check the configured helper for a hostname
+	hc := gitcredentials.HelperConfig{
+		SelfExecutablePath: "/path/to/gh",
+		GitClient:          &git.Client{},
+	}
+	helper, err := hc.ConfiguredHelper("github.com")
+
+	// Then the helper should be configured, but not ours
+	require.NoError(t, err)
+	require.True(t, helper.IsConfigured(), "expected no helper to be configured")
+	require.False(t, helper.IsOurs(), "expected the helper not to be ours but was %q", helper.Cmd)
+}
+
+func TestHelperIsOurs(t *testing.T) {
+	tests := []struct {
+		name string
+		cmd  string
+		want bool
+	}{
+		{
+			name: "blank",
+			cmd:  "",
+			want: false,
+		},
+		{
+			name: "invalid",
+			cmd:  "!",
+			want: false,
+		},
+		{
+			name: "osxkeychain",
+			cmd:  "osxkeychain",
+			want: false,
+		},
+		{
+			name: "looks like gh but isn't",
+			cmd:  "gh auth",
+			want: false,
+		},
+		{
+			name: "ours",
+			cmd:  "!/path/to/gh auth",
+			want: true,
+		},
```

Comment:
**Should we also have a case for Windows `gh.exe`?**

https://github.com/cli/cli/blob/64d50e6fbc4438c2e07b65e5509fc55d7db1f75a/pkg/cmd/auth/shared/gitcredentials/helper_config.go#L77-L89

```suggestion
		{
			name: "ours",
			cmd:  "!/path/to/gh auth",
			want: true,
		},
		{
			name: "ours - Windows edition",
			cmd:  `!"C:\path\to\GitHub CLI\gh.exe" auth`,
			want: true,
		},
```

Reply:
Because Windows paths generally have spaces in them including our own, thought going for the complicated version made most sense.

Reply:
Addressed in https://github.com/cli/cli/pull/9089/commits/3803fd04b9150e40bb5a573d28f562e712cf0ca4